athena/layers/layer.py

```diff
@@ -0,0 +1,116 @@
+"""Layer and WeightedLayer."""
+
+
+class Layer(object):
+    """Network layer."""
+    def __init__(self):
+        self.output = None
```

Comment:
Some layers (namely Dropout) behave differently depending on whether they are used for training or for normal evaluation. Therefore, each layer must have two potentially different [in/out]puts.